Client/Frontend/Brave Notifications/Brave Wallet/WalletConnectionView.swift

```diff
@@ -24,39 +29,120 @@ class WalletConnectionView: UIControl {
     result.contentMode = .scaleAspectFit
     result.setContentHuggingPriority(.required, for: .horizontal)
     result.setContentCompressionResistancePriority(.required, for: .horizontal)
+    result.setContentCompressionResistancePriority(.required, for: .vertical)
     return result
   }()
 
   private let titleLabel: UILabel = {
     let result = UILabel()
     result.textColor = .white
-    result.font = .preferredFont(forTextStyle: .subheadline, weight: .semibold)
-    result.adjustsFontForContentSizeCategory = true
+    result.font = .preferredFont(forTextStyle: .subheadline, weight: .regular)
+    result.adjustsFontForContentSizeCategory = false
     result.numberOfLines = 0
     result.text = Strings.Wallet.dappsConnectionNotificationTitle
     result.setContentCompressionResistancePriority(.required, for: .horizontal)
+    result.setContentCompressionResistancePriority(.required, for: .vertical)
+    if #available(iOS 15, *) {
+      result.adjustsFontForContentSizeCategory = true
+      result.maximumContentSizeCategory = .accessibilityMedium
+    }
     return result
   }()
+  
+  private var cancellable: AnyCancellable?
+  
+  var origin: URLOrigin
+  
+  init(origin: URLOrigin) {
+    self.origin = origin
+    super.init(frame: .zero)
+    setup()
+  }
 
-  override init(frame: CGRect) {
-    super.init(frame: frame)
-
-    addSubview(stackView)
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+      fatalError()
+  }
+  
+  private func setup() {
+    addSubview(scrollView)
+    scrollView.snp.makeConstraints {
+      $0.edges.equalToSuperview()
+    }
+    
+    scrollView.addSubview(stackView)
     stackView.snp.makeConstraints {
       $0.edges.equalToSuperview().inset(24)
     }
     stackView.addArrangedSubview(iconImageView)
     stackView.addArrangedSubview(titleLabel)
     
+    scrollView.contentLayoutGuide.snp.makeConstraints {
+      $0.width.equalTo(self)
+      $0.top.bottom.equalTo(stackView).inset(24)
+    }
+    
     iconImageView.snp.makeConstraints {
       $0.width.height.equalTo(20)
     }
+    
+    snp.makeConstraints {
+      $0.height.equalTo(stackView).inset(-24).priority(.low)
+      $0.height.lessThanOrEqualTo(UIScreen.main.bounds.height / 2)
```

Comment:
`UIScreen.main` is actually deprecated in iOS 16+, we should probably avoid using it directly from now on.  You can try doing `$0.height.equalToSuperview().dividedBy(2)` but it may mean having to wait to attach that constraint until the view is added to the super view